code/game/turfs/turf.dm

```diff
@@ -350,41 +343,43 @@
 
 //////Assimilate Air//////
 /turf/simulated/proc/Assimilate_Air()
-	if(air)
-		var/aoxy = 0 //Holders to assimilate air from nearby turfs
-		var/anitro = 0
-		var/aco = 0
-		var/atox = 0
-		var/asleep = 0
-		var/ab = 0
-		var/atemp = 0
-
-		var/turf_count = 0
-
-		for(var/turf/T in atmos_adjacent_turfs)
-			if(isspaceturf(T))//Counted as no air
-				turf_count++//Considered a valid turf for air calcs
-				continue
-			else if(isfloorturf(T))
-				var/turf/simulated/S = T
-				if(S.air)//Add the air's contents to the holders
-					aoxy += S.air.oxygen
-					anitro += S.air.nitrogen
-					aco += S.air.carbon_dioxide
-					atox += S.air.toxins
-					asleep += S.air.sleeping_agent
-					ab += S.air.agent_b
-					atemp += S.air.temperature
-				turf_count++
-		air.oxygen = (aoxy / max(turf_count, 1)) //Averages contents of the turfs, ignoring walls and the like
-		air.nitrogen = (anitro / max(turf_count, 1))
-		air.carbon_dioxide = (aco / max(turf_count, 1))
-		air.toxins = (atox / max(turf_count, 1))
-		air.sleeping_agent = (asleep / max(turf_count, 1))
-		air.agent_b = (ab / max(turf_count, 1))
-		air.temperature = (atemp / max(turf_count, 1))
-		if(SSair)
-			SSair.add_to_active(src)
+	if(blocks_air)
+		return
+	var/aoxy = 0 //Holders to assimilate air from nearby turfs
+	var/anitro = 0
+	var/aco = 0
+	var/atox = 0
+	var/asleep = 0
+	var/ab = 0
+	var/atemp = 0
+
+	var/turf_count = 0
+
+	for(var/turf/T in atmos_adjacent_turfs)
```

Comment:
```suggestion
	for(var/turf/T as anything in atmos_adjacent_turfs)
```
A turf shouldn't ever actually be null because byond just completely ignores gc rules for them. It just changes source